### ISSUE TITLE
Multiple definitions of `disassembler` and `compiler`

### ISSUE THREAD

**Issue by pcotret** [QDucasse/sdvc#2]
Had some compilation issues. Managed to make it compile with the following changes:
`src/compiler.c`:
```diff
@@ -12,7 +12,8 @@
 #include "table.h"
 #include "value.h"
 
-extern Disassembler* disassembler;
+Disassembler* disassembler;
+Compiler* compiler;
 
 /* Parser structure */
 typedef struct {
@@ -103,7 +104,6 @@ static bool isBinOp(Token* tokenOperator) {
 
 /* Compiler initialization */
 void initCompiler() {
-  compiler = ALLOCATE_OBJ(Compiler);
   compiler->chunk     = initChunk();
   compiler->globals   = initTable();
   compiler->registers = ALLOCATE_ARRAY(Register, REG_NUMBER);
```
`src/compiler.h:`
```diff
@@ -26,7 +26,7 @@ typedef struct {
 } Compiler;
 
 /* Compiler singleton */
-Compiler* compiler;
+extern Compiler* compiler;
```
`src/disassembler.h` 
```diff
@@ -20,7 +20,7 @@ typedef struct {
   FILE* outstream;
 } Disassembler;
 
-Disassembler* disassembler;
+extern Disassembler* disassembler;
```
Not sure if it's the best solution, maybe something wrong in my environment.

**Comment by pcotret** [comment]
Log of the error:
```bash
usr/bin/ld: CMakeFiles/sdvc.dir/src/compiler.c.o:/home/pascal/Documents/ensta/work/sdvc1/src/disassembler.h:23: multiple definition of `disassembler'; CMakeFiles/sdvc.dir/src/chunk.c.o:/home/pascal/Documents/ensta/work/sdvc1/src/disassembler.h:23: first defined here
/usr/bin/ld: CMakeFiles/sdvc.dir/src/disassembler.c.o:/home/pascal/Documents/ensta/work/sdvc1/src/disassembler.h:23: multiple definition of `disassembler'; CMakeFiles/sdvc.dir/src/chunk.c.o:/home/pascal/Documents/ensta/work/sdvc1/src/disassembler.h:23: first defined here
/usr/bin/ld: CMakeFiles/sdvc.dir/src/main.c.o:/home/pascal/Documents/ensta/work/sdvc1/src/disassembler.h:23: multiple definition of `disassembler'; CMakeFiles/sdvc.dir/src/chunk.c.o:/home/pascal/Documents/ensta/work/sdvc1/src/disassembler.h:23: first defined here
/usr/bin/ld: CMakeFiles/sdvc.dir/src/main.c.o:/home/pascal/Documents/ensta/work/sdvc1/src/compiler.h:29: multiple definition of `compiler'; CMakeFiles/sdvc.dir/src/compiler.c.o:/home/pascal/Documents/ensta/work/sdvc1/src/compiler.h:29: first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/sdvc.dir/build.make:238: sdvc] Error 1
make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/sdvc.dir/all] Error 2
make: *** [Makefile:103: all] Error 2
```

**Comment by QDucasse** [comment]
Wrong extern definitions, should be corrected in 38c23db835a6721a7b3d26468e046f158c86e134 !
